### PR TITLE
Move CURAND into CuArrays

### LINKED ISSUE #9: Native rand
Currently I have to do `cu(rand(...))`. It would be nice to have a native version.

### PR DESCRIPTION
**Pull request by andreasnoack**
Fixes #9. Tests pass but there might be some more cleanup to do.

### CODE REVIEW

File: src/rand/wrappers.jl [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+           "CURAND_STATUS_INITIALIZATION_FAILED",
+         CURAND_STATUS_ARCH_MISMATCH => "CURAND_STATUS_ARCH_MISMATCH",
+         CURAND_STATUS_INTERNAL_ERROR => "CURAND_STATUS_INTERNAL_ERROR")
+
+function error_message(status::UInt32)
+    return "cuRAND operation failed with status: " * MESSAGE_BY_STATUS[status]
+end
+
+
+function statuscheck(status::UInt32)
```
Comment (maleadt) [inline src/rand/wrappers.jl]: we've moved to a more sane macro+exception based error handling with the other wrappers, see eg. https://github.com/JuliaGPU/CuArrays.jl/blob/master/src/fft/error.jl

File: src/rand/core.jl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+using CUDAdrv
+ 
+include("init.jl")
+include("defs.jl")
+include("wrappers.jl")
+include("hl.jl")
+
+# println("At the end of module loading: $_rng")
```
Comment (maleadt) [inline src/rand/core.jl]: ?

File: src/rand/hl.jl [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# uniform
```
Comment (maleadt) [inline src/rand/hl.jl]: maybe rename to `highlevel.jl` as with the BLAS wrapper

**Comment by maleadt** [comment]
some minor comments. maybe it would make sense to have clang.jl-based wrappers for the api, such that at least all functions are wrapper (cfr. cuBLAS).

File: src/rand/init.jl [hunk 4]
```diff
@@ -0,0 +1,19 @@
+
+@static if is_windows()
```
Comment (maleadt) [inline src/rand/init.jl]: all this shouldn't be necessary. we have functionality in CUDAapi for this.

**Comment by andreasnoack** [comment]
Thanks for the comments. I think I've managed to address them except for the Clang part. I think it should be okay to do later. Could you restart the tests?

**Comment by maleadt** [comment]
> I think it should be okay to do later.

yeah definitely

> Could you restart the tests?

seems like the open/close trick worked. you can also just log in on the CI page, everybody with JuliaGPU membership has access to that.

**Pull request merged**